src/stability_sdk/animation.py

```diff
@@ -400,69 +375,50 @@ def render_animation(
                             #blur_ks=...,
                             engine_id=self.inpaint_engine_id,
                         )
-                    inpaint_mask = None
 
             # either run diffusion or emit an inbetween frame
-            if (frame_idx-diffusion_cadence_ofs) % diffusion_cadence == 0: # DMARX: i don't thinks subtracting the 'offset' is needed here, it's a modulo already
-
-                # didn't we already do this in self.build_prior_frame_transforms ?
-                # TO DO: add a color match op after each inpaint step.
-                # maybe this just needs to be a subprocedure of image_inpaint?
-                if inpaint_mask is not None:
-                    prior_frames[-1] = image_inpaint(
-                        stub=stub,
-                        image=prior_frames[-1],
-                        mask=inpaint_mask,
-                        prompts=prompts,
-                        weights=weights,
-                        steps=steps//2,
-                        seed=seed,
-                        cfg_scale=args.cfg_scale,
-                        #blur_ks=...,
-                        engine_id=self.inpaint_engine_id,
-                    )
-                    inpaint_mask = None
+            if (frame_idx - self.diffusion_cadence_ofs) % diffusion_cadence == 0:
                 strength = self.frame_args.strength_series[frame_idx]
 
                 # apply additional noising and color matching to previous frame to use as init
-                init_image = prior_frames[-1] if len(prior_frames) and strength > 0 else None
+                init_image = self.prior_frames[-1] if len(self.prior_frames) and strength > 0 else None
                 if init_image is not None:
                     noise = self.frame_args.noise_series[frame_idx]
                     brightness = self.frame_args.brightness_series[frame_idx]
                     contrast = self.frame_args.contrast_series[frame_idx]
                     mix_in = self.frame_args.video_mix_in_series[frame_idx]
 
                     ops = [] # if we previously populated ops before, looks like we're going to overwrite it here. is that on purpose? guessing it's not... I think maybe this should have a different name to distinguish it as init_image specific ops.
```

Comment:
@pharmapsychotic thoughts on this comment?

Reply:
oh it's a totally different set of ops from the prior frame transform ops. this is for prep of init image for generation of next frame.

Reply:
I renamed to init_ops to disambiguate 👍